neqo-transport/src/packet/mod.rs

```diff
@@ -917,14 +928,58 @@ impl fmt::Debug for Public<'_> {
     }
 }
 
-pub struct Decrypted<'a> {
+/// Error information from a failed decryption attempt.
+/// Contains minimal packet information needed for error handling.
+#[derive(Debug)]
+pub struct DecryptionError {
+    /// The error that occurred
+    pub error: Error,
+    /// The original packet data (unchanged since decryption failed)
+    pub data: Vec<u8>,
+    /// The destination connection ID
+    pub dcid: ConnectionId,
+    /// The packet type
+    pub packet_type: Type,
+}
+
+impl From<(&Public<'_>, Error)> for DecryptionError {
+    fn from((packet, error): (&Public, Error)) -> Self {
+        Self {
+            error,
+            data: packet.data.to_vec(),
+            dcid: packet.dcid.clone(),
+            packet_type: packet.packet_type,
```

Comment:
It would be nice if this code did nothing if logging was not enabled.

Reply:
We can make that so once we have fixed #1894. Let me take a stab at that.

Reply:
I just looked at this again, and I think I lost context. What was your concern here @martinthomson?

Reply:
So, with #1894, we won't do anything with logs, but this error will be constructed when an error occurs.  That construction involves a bunch of work that will be thrown out.

It's only error cases, so we probably won't be wasting too much effort overall, so don't contort the code too badly if it doesn't work out, but it seems like you could add an Option somewhere.